proxyserver/middleware/s3api.go

```diff
@@ -314,6 +317,41 @@ func s3PathSplit(path string) (string, string) {
 	}
 }
 
+func validBucketName(s string) bool {
+	if len(s) < 3 || len(s) > 63 {
+		return false
+	}
+	if !(unicode.IsLetter(rune(s[0])) || unicode.IsDigit(rune(s[0]))) {
+		return false
+	}
+	if strings.Index(s, ".-") != -1 {
+		return false
+	}
+	if strings.Index(s, "-.") != -1 {
+		return false
+	}
+	if strings.Index(s, "..") != -1 {
+		return false
+	}
+	if s[len(s)-1] == '.' {
+		return false
+	}
+	ip := regexp.MustCompile(`(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)(\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)){3}`)
```

Comment:
Can these regexes be global vars so it doesn't have to compile them every request?